### DISCUSSION TITLE
div added to HTML uib-element

### DISCUSSION THREAD

**Discussion by buschhardt** [TotallyInformation/node-red-contrib-uibuilder#210] [Q&A] [answered]
### Software and Package Versions
Software       | Version
-------------- | -------
Node.JS        |  18.15.0
npm            |  9.6.4
Node-RED       |  3.0.2
uibuilder node |  6.4.1
uibuilderFE    |  ? (Where can I find this?) No framework, IIEF client
OS             |  Debian 5.10.162-1
Browser        |  Chromium 112.0.5615.49

### How is Node-RED installed? Where is uibuilder installed?

nodered installed locally
uibuilder installed by palettesystem in nodered

### Question

When I add a payload to HTML.uib-elemnt to uibuilder, adds a `<div></div>` around my payload. Is that correct? If yes its possible to modify, so that I can use my own tags?
Thats my tag
```
<a-scene>
</a-scene>
```
after adding the payload `<a-asset>This is the payload: 1681985730455 !</a-asset>`

I get

```
<a-scene>
<div><a-asset>This is the payload: 1681985730455 !</a-asset></div>
</a-scene>
```
Thank You

**Comment by TotallyInformation** [comment]
Hi, moved this over to discussions.

Yes, `uib-element` always wraps in a `<div>` tag which carries the ID that you define. This cannot currently be changed within `uib-element` itself though it could be potentially added to the backlog.

There is, however, a relatively easy work-around. Because the output of the node is just configuration data that the uibuilder client library converts to HTML, you can make your own amendments to the output before sending it to the uibuilder node.

For example, if using the simple list output, the wrapper `div` is defined at `msg._ui[0].components[0].type` and has the value `div`. So if you wanted to change that to a different block tag, you can add a change node like this:

![image](https://user-images.githubusercontent.com/1591850/233342862-62c8ebc8-a3d9-4da3-95bf-f5f888d2f76f.png)

You can get the path to the correct property to change from a debug node which has a "copy path" button on each entry.

This is one of the many benefits of working via a standard schema since it makes amendments easy and is very flexible without having to dive into the many quirks of HTML itself.

I will add the ability to change the wrapper tag more directly to the uibuilder roadmap.

**Reply by TotallyInformation** [reply, marked as answer]
BTW, the tag value could, of course be a different property of the msg you send through `uib-element` since other properties are passed through.

![image](https://user-images.githubusercontent.com/1591850/233344092-bd272a8e-ee82-4c48-b1ef-e42f8e83d6e8.png)

**Reply by buschhardt** [reply]
Thank You, the workaround is good.
